sdk/python/test/e2e/utils.py

```diff
@@ -17,15 +17,14 @@ def verify_unschedulable_job_e2e(
     logging.info(f"\n\n\n{job_kind} is creating")
     client.wait_for_job_conditions(name, namespace, job_kind, {constants.JOB_CONDITION_CREATED})
 
-    # Job should have Created conditions.
-    conditions = client.get_job_conditions(name, namespace, job_kind)
-    if len(conditions) != 1:
-        raise Exception(f"{job_kind} conditions are invalid: {conditions}")
-
-    # Job should have correct conditions.
+    # Job should have a Created condition.
     if not client.is_job_created(name, namespace, job_kind):
         raise Exception(f"{job_kind} should be in Created condition")
 
+    # Job shouldn't have a Running condition.
+    if client.is_job_running(name, namespace, job_kind):
```

Comment:
Will this be flaky? What if job  gets into running state?

Reply:
In that case, the test should be failed since we pass an unschedulable job to `def verify_unschedulable_job_e2e`.

https://github.com/kubeflow/training-operator/blob/0f292f4216e9fdf2e000cd33fc024bd896236b18/sdk/python/test/e2e/test_e2e_tfjob.py#L64

Reply:
ping @johnugeorge

Reply:
@tenzen-y I agree with the behavior, that if we're passing an unschedulable job and it has a running state then our test should fail.
My doubt is, are we sure that the job we're passing is unschedulable? Because in my testing the job had two states (created and running). If the job is unschedulable then it shouldn't have running state, right?

Reply:
> My doubt is, are we sure that the job we're passing is unschedulable?

@nagar-ajay Ah, I see.

In the previous test, we verify the number of Conditions. So, If the Job has `Created=true` and `Running=false`, this test unintentionally fails.

In this PR, the test would be passed in that case (`Created=true` and `Running=false`).

Does that make sense?

Reply:
yup, that makes sense.

Reply:
Let me summarize for @johnugeorge.

In the previous Implementation, tests go to fail if the Job has `Created=false OR Has(Running)`.

In this implementation,  tests go to fail only if Job has `Created=false OR Running=true`.